str0m/src/media/mline.rs

```diff
@@ -446,6 +410,81 @@ impl MLine {
         })
     }
 
+    pub fn create_padding_packet(
+        &mut self,
+        target_size: DataSize,
+        now: Instant,
+        exts: &Extensions,
+        twcc: &mut u64,
+        buf: &mut Vec<u8>,
+    ) -> Option<(RtpHeader, SeqNo)> {
```

Comment:
Rather than this being a `pub fn` by itself, I like the code path to go via `poll_packet`. In Session, you already have the distinction that "if there is a padding length, do this, otherwise do that". We just move that down one level, so `poll_packet` gets another argument `pad_len: Option<u8>`.

Then we don't need to break out `finalize_packet`, but keep that code intact (less git diff).

Finally, yes, as you say, i think the padding packet creation should be pushed down into the `rtp` subcrate.